source/components/IDInput/index.jsx

```diff
@@ -85,14 +83,17 @@ const IDInput = ({
   };
 
   const handleChangeContactName = (e) => {
-    setContactName(e.target.value);
+    const name = e.target.value;
+    const existingContact = contacts.find((contact) => contact.name === name);
```

Comment:
should we also check if the `contact.id === name`? in case the user wants to name his contact `plug.icp` for example 🤔

Reply:
there's no duplication in that case, but maybe its confusing